frontend/web/components/modals/CreateSegment.tsx

```diff
@@ -420,6 +418,7 @@ const CreateSegment: FC<CreateSegmentType> = ({
             entity={segmentContentType?.model}
             onChange={(m) => {
               setMetadata(m as Metadata[])
+              // Need to fix this to be more robust and handle post save
```

Comment:
This part is responsible for the false signal of data being edited. It's happening on prod too and the logic is too nested to be quick fixed.